R/calcFloorspacePast.R

```diff
@@ -0,0 +1,128 @@
+#' Historical residential floor space demand
+#'
+#' Data for many European countries is taken from EEA, India from IEA and other
+#' countries from Daioglou et al 2012. The result does not cover all countries
+#' and has mixed points in time depending on the region.
+#'
+#' @note RK: In Antoine's EDGE-B, data points associated with an GDP/POP above
+#' 70000 USD/cap are dropped here to improve the later regression. This
+#' filtering should be moved to getFloorspaceResidential where the regression is
+#' performed. Therefore, the high-income data points are kept at this stage.
+#'
+#' @returns MAgPIE object with historic floor space
+#'
+#' @author Robin Krekeler, Antoine Levesque
+#'
+#' @importFrom madrat readSource calcOutput
+#' @importFrom quitte as.quitte calc_addVariable
+#' @importFrom dplyr filter mutate select anti_join group_by left_join
+#' @importFrom rlang .data
+#' @importFrom magclass mbind as.magpie collapseDim
+#' @importFrom tidyr spread
+#' @export
+
+calcFloorspacePast <- function() {
+
+  # FUNCTIONS ------------------------------------------------------------------
+
+  # specific floor space from absolute floor space and population
+  floorPerCap <- function(floor, pop) {
+    floor %>%
+      rbind(pop) %>%
+      calc_addVariable("specific floor space" = "`floor space` / population",
+                       units = "m2/cap", only.new = TRUE) %>%
+      mutate(demographic = "Total")
+  }
+
+
+  # LOAD AND CALCULATE DATA ----------------------------------------------------
+
+  # data from Daioglou et al.
+  daioglou <- readSource("Daioglou") %>%
+    as.quitte(na.rm = TRUE) %>%
+    filter(.data[["quintile"]] == 0) %>%
+    select(-"quintile") %>%
+    mutate(variable = "specific floor space",
+           unit = "m2/cap")
+
+  # drop ESP and PRT from EEA data (to high uncertainty)
+  eea <- readSource("EEAfloorspace") %>%
+    as.quitte(na.rm = TRUE) %>%
+    filter(!.data[["region"]] %in% c("ESP", "PRT")) %>%
+    mutate(variable = "floor space",
+           unit = "million m2")
+
+  # take only India from IEA
+  ind <- readSource("IEAfloorspace", convert = FALSE) %>%
```

Comment:
Maybe in the future we want to write a convert function that should apply everywhere, so maybe it's better to remove `convert = FALSE` here?